src/devtools/client/debugger/src/actions/sources/newSources.ts

```diff
@@ -195,20 +206,26 @@ export function newOriginalSources(sourceInfo: SourceData[]): UIThunkAction<Prom
     for (const source of sources) {
       dispatch(checkPendingBreakpoints(cx, source.id));
     }
+    */
 
     return sources;
   };
 }
 
-export function newGeneratedSource(sourceInfo: SourceData): UIThunkAction<Promise<Source>> {
+export function newGeneratedSource(sourceInfo: SourceData): UIThunkAction<Promise<SourceDetails>> {
   return async dispatch => {
     const sources = await dispatch(newGeneratedSources([sourceInfo]));
     return sources[0];
   };
 }
-export function newGeneratedSources(sourceInfo: SourceData[]): UIThunkAction<Promise<Source[]>> {
+export function newGeneratedSources(
+  sourceInfo: SourceData[]
+): UIThunkAction<Promise<SourceDetails[]>> {
   return async (dispatch, getState, { ThreadFront }) => {
     const resultIds: string[] = [];
+
+    // TODO Rewrite "check pending" handling
```

Comment:
I suspect that this code will end up getting deleted eventually, but we need to replace it with something else to do the job it was doing (select the previously selected location and recreate previous breakpoints once sources are loaded).